From eb0c51bff2aa4bda6c31a83f489dce9b39afddac Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Thu, 27 Mar 2014 12:59:24 +0100
Subject: [PATCH 01/56] KCM: Remove developer mode

This also means that source code is now always indexed. A normal user
typically won't have any source code and the developers can always
exclude that folder from the list of excluded folder. It's way way more
efficient. Mimetype filtering is expensive.

This speeds up the first level indexing.
---
 src/file/fileexcludefilters.cpp |  3 +--
 src/file/kcm/configwidget.ui    |  9 +--------
 src/file/kcm/kcm.cpp            | 14 --------------
 src/file/kcm/kcm.h              |  1 -
 4 files changed, 2 insertions(+), 25 deletions(-)

diff --git a/src/file/fileexcludefilters.cpp b/src/file/fileexcludefilters.cpp
index 1888543..6f31a06 100644
--- a/src/file/fileexcludefilters.cpp
+++ b/src/file/fileexcludefilters.cpp
@@ -166,8 +166,7 @@ QStringList Baloo::sourceCodeMimeTypes()
 
 QStringList Baloo::defaultExcludeMimetypes()
 {
-    // By default we want to index everything apart from source code
-    return sourceCodeMimeTypes();
+    return QStringList();
 }
 
 int Baloo::defaultExcludeMimetypesVersion()
diff --git a/src/file/kcm/configwidget.ui b/src/file/kcm/configwidget.ui
index 32b0b6b..484ec2c 100644
--- a/src/file/kcm/configwidget.ui
+++ b/src/file/kcm/configwidget.ui
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <ui version="4.0">
- <author>Sebastian Trueg</author>
+ <author>Vishesh Handa</author>
  <class>ConfigWidget</class>
  <widget class="QWidget" name="ConfigWidget">
   <property name="geometry">
@@ -55,13 +55,6 @@
     </widget>
    </item>
    <item>
-    <widget class="QCheckBox" name="m_checkboxSourceCode">
-     <property name="text">
-      <string>Enable Developer Mode</string>
-     </property>
-    </widget>
-   </item>
-   <item>
     <spacer name="bottomSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
diff --git a/src/file/kcm/kcm.cpp b/src/file/kcm/kcm.cpp
index 7e83210..354392b 100644
--- a/src/file/kcm/kcm.cpp
+++ b/src/file/kcm/kcm.cpp
@@ -69,8 +69,6 @@ ServerConfigModule::ServerConfigModule(QWidget* parent, const QVariantList& args
 
     connect(m_folderSelectionWidget, SIGNAL(changed()),
             this, SLOT(changed()));
-    connect(m_checkboxSourceCode, SIGNAL(toggled(bool)),
-            this, SLOT(changed()));
 }
 
 
@@ -89,10 +87,6 @@ void ServerConfigModule::load()
     QStringList excludeFolders = group.readPathEntry("exclude folders", QStringList());
     m_folderSelectionWidget->setFolders(includeFolders, excludeFolders);
 
-    // MimeTypes
-    QStringList mimetypes = config.group("General").readEntry("exclude mimetypes", defaultExcludeMimetypes());
-
-    m_oldExcludeMimetypes = mimetypes;
     m_oldExcludeFolders = excludeFolders;
     m_oldIncludeFolders = includeFolders;
 
@@ -117,12 +111,6 @@ void ServerConfigModule::save()
     config.group("General").writePathEntry("folders", includeFolders);
     config.group("General").writePathEntry("exclude folders", excludeFolders);
 
-    QStringList excludeMimetypes;
-    if (m_checkboxSourceCode->isChecked())
-        excludeMimetypes = sourceCodeMimeTypes();
-
-    config.group("General").writeEntry("exclude mimetypes", excludeMimetypes);
-
     // Start Baloo
     if (indexingEnabled) {
         const QString exe = KStandardDirs::findExe(QLatin1String("baloo_file"));
@@ -143,8 +131,6 @@ void ServerConfigModule::save()
         cleaningRequired = true;
     else if (excludeFolders != m_oldExcludeFolders)
         cleaningRequired = true;
-    else if (excludeMimetypes != m_oldExcludeMimetypes)
-        cleaningRequired = true;
 
     if (cleaningRequired) {
         const QString exe = KStandardDirs::findExe(QLatin1String("baloo_file_cleaner"));
diff --git a/src/file/kcm/kcm.h b/src/file/kcm/kcm.h
index 2e7a534..ae76af4 100644
--- a/src/file/kcm/kcm.h
+++ b/src/file/kcm/kcm.h
@@ -42,7 +42,6 @@ public Q_SLOTS:
 private:
     QStringList m_oldIncludeFolders;
     QStringList m_oldExcludeFolders;
-    QStringList m_oldExcludeMimetypes;
 };
 }
 
-- 
1.9.1

